### DISCUSSION TITLE
Recommendation Ideas

### DISCUSSION THREAD

**Discussion by altanborali16** [altanborali16/SWE574-Group2#137] [Ideas]
Lets say I have communities that I subscribed and I am getting all their categories as list.
Lets say i have 20 categories.
I take a loop on communities which are not subscribed by me and public, and check their categories.

Try to find from those communities if they have any match.

IF THERE IS A MATCH

Get the posts and randomly sort them by date by like count etc. so each time user refreshes page user have different sorting.

IF THERE IS NO MATCH

Go to communities and take newest from each.

@kotobusdriver

**Comment by kotobusdriver** [comment]
The desired list of communities can probably be fetched with a single SQL statement with some nested selects with the mentioned conditions.

About the post selection from among this narrowed down list of communities: Are we planning to display at least one post from each community? Or would it be okay to display the top posts from across all communities we have found?

If we are okay with the latter, then we can just get the top n posts from the like-ordered list of all posts that belong to the communities we have found.
